[nmstate-0.3] nm connection: Only use active connection for VLAN parent UUID

cathay4t opened the pull request:
When there are multiple NM profiles exists, nmstate might choose the
inactive profile as VLAN parent which lead to error:

    libnmstate.error.NmstateLibnmError: Activate profile: testbond0.400
    failed: error=nm-manager-error-quark: Failed to find a compatible device
    for this connection (3)

The fix is preferring profile with active connection when choosing
VLAN/VxLAN parent.

ffmancera reviewed (dismissed):
IMO, the fix is ugly but well, in nmstate-0.3 the whole activation/profile code is ugly so.. LGTM. Thanks!

ffmancera reviewed (changes requested):
Please, could you check why is this failing on the ovs test? I think it was not failing before.. does this happen only on CI? If so let's merge it anyway..

cathay4t commented:
> Please, could you check why is this failing on the ovs test? I think it was not failing before.. does this happen only on CI? If so let's merge it anyway..

Nice cache. I have fixed it with new patch.

ffmancera commented on libnmstate/nm/applier.py:
```diff
@@ -693,9 +693,23 @@ def _use_uuid_for_parent(context, pending_con_profiles, save_to_disk):
+        if (
+            cur_nm_profile.get_interface_name()
+            not in kernel_iface_name_to_uuid
+        ) and connection_type not in (
```
This is incorrect. Here connection_type has not been assigned.

```
E           UnboundLocalError: local variable 'connection_type' referenced before assignment
```

cathay4t replied:
Thanks. Fixed.

cathay4t committed:
nm connection: Only use active connection for VLAN parent UUID

When there are multiple NM profiles exists, nmstate might choose the
inactive profile as VLAN parent which lead to error:

    libnmstate.error.NmstateLibnmError: Activate profile: testbond0.400
    failed: error=nm-manager-error-quark: Failed to find a compatible device
    for this connection (3)

The fix is preferring profile with active connection when choosing
VLAN/VxLAN parent.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
